src/lib.rs

```diff
@@ -135,23 +135,21 @@ where
     }
 
     pub fn check_data_ready(&mut self) -> Result<(), Error<E>> {
-        let mut retries = 10;
+        let mut retries = 20;
         while retries > 0 {
             let status = self.read_register(Register::ST1)?;
             if (status & 0x01) != 0 {
                 return Ok(()); // Data ready
             }
-            std::thread::sleep(std::time::Duration::from_millis(100));
             retries -= 1;
         }
         Err(Error::DataNotReady)
     }
     // 9.4.3.2. Normal Read Sequence:
     //   1. Check Data Ready or not by any of the following method:
     //      |ST2| -> | 0 0 0 0 HOFL INV 0 0 |
-    pub fn check_overflow(&mut self) -> Result<(), Error<E>> {
-        let status = self.read_register(Register::ST2)?;
-        if (status & 0x04) != 0 {
+    pub fn check_overflow(&mut self, register: u8) -> Result<(), Error<E>> {
```

Comment:
This should be in [3834c31fed403f475f6b003e270c234eecbfaadb](https://github.com/bluerobotics/AK09915-rs/commit/3834c31fed403f475f6b003e270c234eecbfaadb)

Reply:
@patrickelectric